### ISSUE TITLE
aws_fsx_ontap_file_system incorrectly requires two subnet ids when used with single AZ

### ISSUE THREAD

**Issue by RansomingSql** [hashicorp/terraform-provider-aws#25339]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.2.2
on windows_amd64
+ provider registry.terraform.io/hashicorp/aws v3.75.2
+ provider registry.terraform.io/hashicorp/random v3.3.0

### Affected Resource(s)

* aws_fsx_ontap_file_system

### Terraform Configuration Files

```hcl
variable "subnet_1" {
  default = "subnet-*" # Supply your own subnet ids
}

variable "subnet_2" {
  default = "subnet-*"# Supply your own subnet ids
}

terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
    }
  }
}



resource "random_password" "netapp-fsxn-password" {
  special          = true
  override_special = "!#$%&*()-_=+[]{}<>:?"
  length           = 20
  min_lower        = 1
  min_numeric      = 1
  min_special      = 1
  min_upper        = 1
}

resource "aws_fsx_ontap_file_system" "test" {
  storage_capacity                  = 1024
  #subnet_ids                        = [var.subnet_1, var.subnet_2]
  subnet_ids                        = [var.subnet_1]
  deployment_type                   = "SINGLE_AZ_1"
  throughput_capacity               = 512
  preferred_subnet_id               = var.subnet_1
  fsx_admin_password                = random_password.netapp-fsxn-password.result
  automatic_backup_retention_days   = 0
}

```

### Debug Output
https://gist.github.com/RansomingSql/c364b83d2c8e96f7e83d36f715dd385a

### Expected Behavior

The terraform resource 'aws_fsx_ontap_file_system' should only expect one subnet when 'deployment_type' = 'SINGLE_AZ_1'
```hcl
resource "aws_fsx_ontap_file_system" "test" {
  storage_capacity                  = 1024
  #subnet_ids                        = [var.subnet_1, var.subnet_2]
  subnet_ids                        = [var.subnet_1]
  deployment_type                   = "SINGLE_AZ_1"
  throughput_capacity               = 512
  preferred_subnet_id               = var.subnet_1
  fsx_admin_password                = random_password.netapp-fsxn-password.result
  automatic_backup_retention_days   = 0
}
```

### Actual Behavior

The terraform resource throws a parameter validation error when you only specify one subnet as required by AWS API. Error is 
│ Error: Not enough list items
│
│   with aws_fsx_ontap_file_system.test,
│   on main.tf line 40, in resource "aws_fsx_ontap_file_system" "test":
│   40:   subnet_ids                        = [var.subnet_1]
│
│ Attribute requires 2 item minimum, but config has only 1 declared.

When you specify 2 subnets as required by the terraform resource you get an error from the AWS API of: 

│ Error: error creating FSx ONTAP File System: BadRequest: Exactly 1 subnet ID(s) is required for SINGLE_AZ_1.
│
│   with aws_fsx_ontap_file_system.test,
│   on main.tf line 36, in resource "aws_fsx_ontap_file_system" "test":
│   36: resource "aws_fsx_ontap_file_system" "test" {

### Steps to Reproduce

1. Copy provided configuration to directory
2. Append your own 'provider' information in terraform
3. `terraform apply`

### References

https://docs.aws.amazon.com/cli/latest/reference/fsx/create-file-system.html

**Comment by justinretzolk** [comment]
Hey @RansomingSql 👋 Thank you for taking the time to raise this! It looks like this was corrected in [AWS Provider version `4.11.0`](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4110-april-22-2022):

> resource/aws_fsx_ontap_file_system: Add support for SINGLE_AZ_1 deployment_type. (https://github.com/hashicorp/terraform-provider-aws/issues/24280)

Can you update to at least this version of the provider and verify that that resolves your issue?

**Comment by RansomingSql** [comment]
@justinretzolk , yes that seems to have fixed the problem. Sorry about raising a duplicate issue: I'm still figuring out how to update all the parts of terraform so I missed that one :) Thanks for finding the fixed version.
